File: Search/BinarySearch.java
import java.util.*;
public class BinarySearch{
    public static void main(String[] args) {
        int [] arr = {12, 13, 222, 45, 555 , 689, 90, 4, 38};
        int key = 90;
        int l=0;
        int h= arr.length-1;
        while(l<=h){
            int middle= (l+h)/2;
            if(key>arr[middle]){
                l= middle + 1;
            } else if(key < arr[middle]){
                h= middle - 1;
            }else{
                System.out.println(middle);
                return;
            }
        }
        System.out.println("not found"+key);
    }
}

File: Search/binary_search.py
def binarysearch(arr,key):
    low=0
    high=len(arr)-1

    while(low<=high):
        mid=(low+high)//2
        if(arr[mid]==key):
            return mid
        elif(arr[mid]<key):
            low=mid+1
        else:
            high=mid-1

    return -1

myarray=[1,3,5,9,12,45,8]
key=91
result=binarysearch(myarray,key)

if(result!=-1):
    print("Value",key,"found at index",result)

else:
    print("Key not found in array")    
